### PR TITLE
Add action_sequence, a datum for creating sleep()less actors

### PR DESCRIPTION
**Pull request by Cyberboss**
See action_sequence.dm for documentation

TL;DR: Instead of using sleeps or handmade timers for scripted sequences, you create one of these datums which will handle the logistics of creating timers to link the steps and check pre/postrequisites. You configure it to conditionally determine the next step, use client time timers, and more.

TODO:
- [x] Convert the plushie fight sequence as an example

### CODE REVIEW

**Comment by XDTM** [comment]
Could be good for hallucinations

**Comment by Iamgoofball** [comment]
Todo: Convert fultons

**Comment by kevinz000** [comment]
help i'm too much of a smoothbrain to understand!

**Comment by Cyberboss** [comment]
Instead of using sleeps for scripted sequences, you create one of these datums which will handle the logistics of creating timers to link the steps and check pre/postrequisites.

File: code/datums/action_sequence.dm [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	callback = CALLBACK(src, null)
+	INVOKE_ASYNC(src, .proc/NextStep)
+
+/datum/action_sequence/proc/NextStep()
+	if(QDELING(src))
+		return
+
+	if(!next_step)
+		if(steps_invoked <= 1)
+			WARNING("[type] ran out of steps after 1 or 0 runs. Consider demoting it from an action_sequence")
```
Comment (Cyberboss) [inline code/datums/action_sequence.dm]: debatable warning

Comment (optimumtact) [reply code/datums/action_sequence.dm]: sleep has it's place for action sequences that are trivial

**Comment by Cyberboss** [comment]
TODO: persistent requirements

**Comment by optimumtact** [comment]
todo, not let this get stale

**Comment by Iamgoofball** [comment]
FINISH THIS @cyberboss

**Pull request closed without merging**